### ISSUE TITLE
MacBook Pro 14", M4 Max 16C, 128GB, Sequoia 15.1, Xcode 16.1 [89.621 sec]

### ISSUE THREAD

**Issue by kevingriffin** [devMEremenko/XcodeBenchmark#532]
```
** BUILD SUCCEEDED ** [89.621 sec]

System Version: 15.1
Xcode 16.1
Hardware Overview
      Model Name: MacBook Pro
      Model Identifier: Mac16,6
      Total Number of Cores: 16 (12 performance and 4 efficiency)
      Memory: 128 GB

✅ XcodeBenchmark has completed
1️⃣  Take a screenshot of this window (Cmd + Shift + 4 + Space) and resize to include:
\t- Build Time (See ** BUILD SUCCEEDED ** [XYZ sec])
\t- System Version
\t- Xcode Version
\t- Hardware Overview
\t- Started 15:56:30
\t- Ended   15:58:02
\t- Date 2024年 11月 9日 土曜日 15時58分02秒 JST
```

<img width="752" alt="スクリーンショット 2024-11-09 15 58 22" src="https://github.com/user-attachments/assets/a3dca602-9c60-42e4-94c0-23c7a407bc64">

**Comment by alexandre-g** [comment]
@kevingriffin Thanks for submitting. Did you have high power mode enabled for the run?

**Comment by kevingriffin** [comment]
I did, but I wasn't sure if that was actually appropriate. Let me know and I can do another run if needed.

**Comment by alexandre-g** [comment]
No that's right, it's meant to be on. Just a little surprising that it got almost the same as M3 Max 16-core thats on the results, which is a 16" MBP. Maybe thermal throttling? Do try another run if you can!

**Comment by kevingriffin** [comment]
Agreed, I'll shut it down and let it cool off for a while for another run 👍

**Comment by marvin-hansen** [comment]
Please post an update after a second run.

**Comment by kevingriffin** [comment]
Second run was 87.533, a minor improvement but not what we'd expect from Geekbench.

It's (obviously) new, but Spotlight indexing seems finished and the system sits at 99% idle before the test is run.

Is there anything that lets you monitor for thermal throttling? I know asitop had a lot of these features, but it doesn't seem to work well on more recent Apple Silicon.

<img width="682" alt="スクリーンショット 2024-11-09 19 27 36" src="https://github.com/user-attachments/assets/23c8311f-e6dd-4925-b3ab-697d7797d676">

**Comment by marvin-hansen** [comment]
This might be unrelated to the actual CPU.
The Swift compiler is a frontend to LLVM and I made recently an observation the the latest tw LLVM version 18 and 19 are relatively slow especially. When I switched to a different compiler to build C dependencies for my project (Rust) things were a lot faster.

The recent 30% to 40% regression of Xcode 16 vs 15 indicates that something went seriously wrong with the Apple compiler. For reference, the Rust compiler team never ships any regression in excess of single digit % changes relative to the previous version so I wouldn't draw any conclusion about the hardware yet.

Instead, I think it would be useful to design a different benchmark without XCode where the compiler version is fixed and largely able to exhaust all available cores.

**Comment by alexandre-g** [comment]
I am only surprised as an M4 Max is consistently and by a good margin faster than both M2 Ultra and M3 Max in Geekbench benchmarks, including in a clang one. I haven't checked closely, but benchmarks for other models seem to roughly carry across to the performance differences between hardware in this benchmark, yet not this one

**Comment by alexandre-g** [comment]
The 16" got 79s [here](https://github.com/devMEremenko/XcodeBenchmark/issues/535), which is the time one would expect (for the M4 Max 16c)

@kevingriffin did you run Geekbench benchmark on yours by any chance? Can you share the result run if so?

**Comment by flowridr** [comment]
> No that's right, it's meant to be on. Just a little surprising that it got almost the same as M3 Max 16-core thats on the results, which is a 16" MBP. Maybe thermal throttling? Do try another run if you can!

I'm getting better results (around 3 secs) with my m4 max and auto mode

**Comment by marvin-hansen** [comment]
Would someone please measure temperature when the 14" Max runs the benchmark?

According to this report, the M4 gained a massive 32% in performance per Watt.

https://www.jeffgeerling.com/blog/2024/m4-mac-minis-efficiency-incredible

It might be that the limited thermal design of the 14" model requires thermal throttling to keep everything stable as the M4 peaks now at about 100Watt vs about 80Watt with the M3. 

I loosely remember that thermal throttling of the 14" already came up with the M3 series, but it was way less of an issue.

The 16" model always had more cooling capacity hence can run the CPU longer at higher frequency and therefore delivere better sustained performance.

**Comment by alexandre-g** [comment]
>Is there anything that lets you monitor for thermal throttling? I know asitop had a lot of these features, but it doesn't seem to work well on more recent Apple Silicon.

@kevingriffin this guy says some patch is needed? Has a repo with it in the comments
https://x.com/ivanfioravanti/status/1856110736314958064

https://github.com/ivanfioravanti/asitop

**Comment by marvin-hansen** [comment]
Just a quick question, was the original run made on battery?

Apparently you get slightly better performance when on a power chord, as discussed in this review.

https://youtu.be/eqsuFlT6WoE
